pkg/utils/helmrepo.go

```diff
@@ -538,7 +538,15 @@ func FilterCharts(sub *appv1.Subscription, indexFile *repo.IndexFile) error {
 func takeLatestVersion(indexFile *repo.IndexFile) (err error) {
 	indexFile.SortEntries()
 
-	for k := range indexFile.Entries {
+	// Don't call helm Get to pick up the latest version when there is only one version for the given chart. This is to avoid the error
+	// If there is only one version for the chart and it is defined as `3.0.0-stable`,
+	// the helm Get function fails to identify the version pattern with error
+	for k, chartVersions := range indexFile.Entries {
+		if len(chartVersions) == 1 {
+			klog.Infof("only one version is found, no need to helm Get latest version, chart name: %v, version: %v", k, chartVersions[0].Version)
```

Comment:
```suggestion
			klog.V(2).Infof("only one version is found, no need to helm Get latest version, chart name: %v, version: %v", k, chartVersions[0].Version)
```

Let's avoid changing the default behaviour too much by suppressing this logging a bit.